src/components/SchemaViewer.svelte

```diff
@@ -11,15 +11,32 @@
       .split(" ")
       .filter((t) => t.length > 0);
 
-    const addVisibility = (node) => {
+    const addVisibility = (node, parentNodeNames) => {
```

Comment:
If you wanted to reduce the amount of conditional branches in this function, I think you could set a default value to `parentNodeNames=["root"]` and remove a few of the `parentName.length > 0` checks.

Reply:
Okay